rtlib/src/ImageStitcher.cpp

```diff
@@ -0,0 +1,732 @@
+        // features_[i].keypoints.size());
+    }
+
+    // find features possibly in parallel
+    // if (rois_.empty())
+
+    (*features_finder_)(feature_find_imgs, features_);
+
+    // else
+    //(*features_finder_)(feature_find_imgs, features_, feature_find_rois);
+
+    // Do it to save memory
+    feature_find_imgs.clear();
+    feature_find_masks.clear();
+
+    return features_;
+}
+
+void ImageStitcher::find_matches_(double conf_thresh_, std::vector<cv::UMat>& seam_est_imgs_, std::vector<cv::Size>& full_img_sizes_) {
+    std::vector<cv::detail::MatchesInfo> pairwise_matches_;
+    cv::Ptr<cv::detail::FeaturesMatcher> features_matcher_ = cv::makePtr<cv::detail::AffineBestOf2NearestMatcher>(false, false);
+    cv::UMat matching_mask_;
+    // LOGLN("Finding features, time: " << ((getTickCount() - t) /
+    // getTickFrequency()) << " sec");
+
+    (*features_matcher_)(all_features_, all_pairwise_matches_, matching_mask_);
+    features_matcher_->collectGarbage();
+    // LOGLN("Pairwise matching, time: " << ((getTickCount() - t) /
+    // getTickFrequency()) << " sec");
+
+    // This is for option 1 with placing the matches before filtering the images
+    if (option_ == 1 && !landmarks_.empty()){
+        for(int i = 0; i < landmarks_.size(); i++){
+            insert_user_matches_(landmarks_[i]);
+        }
+    }
+    // Leave only images we are sure are from the same panorama
+    auto indices_ = cv::detail::leaveBiggestComponent(
+            all_features_, all_pairwise_matches_, (float)conf_thresh_);
+    std::vector<cv::UMat> seam_est_imgs_subset;
+    std::vector<cv::UMat> imgs_subset;
+    std::vector<cv::Size> full_img_sizes_subset;
+    for (size_t i = 0; i < indices_.size(); ++i) {
+        imgs_subset.push_back(imgs_[indices_[i]]);
+        seam_est_imgs_subset.push_back(seam_est_imgs_[indices_[i]]);
+        full_img_sizes_subset.push_back(full_img_sizes_[indices_[i]]);
+    }
+    seam_est_imgs_ = seam_est_imgs_subset;
+    imgs_ = imgs_subset;
+    full_img_sizes_ = full_img_sizes_subset;
+
+}
+
+std::vector<cv::detail::CameraParams> ImageStitcher::estimate_camera_params_(double conf_thresh_, float& warped_image_scale_, std::vector<cv::detail::ImageFeatures>& features_,
+                                                                        std::vector<cv::detail::MatchesInfo>& pairwise_matches_) {
+    //////////////////////////////////
+    ///// Estimate camera params /////
+    //////////////////////////////////
+    // estimate homography in global frame
+    cv::Ptr<cv::detail::BundleAdjusterBase> bundle_adjuster_ = cv::makePtr<cv::detail::BundleAdjusterAffinePartial>();
+    cv::Ptr<cv::detail::Estimator> estimator_ =
+            cv::makePtr<cv::detail::AffineBasedEstimator>();
+    std::vector<cv::detail::CameraParams> cameras_;
+    if (!(*estimator_)(features_, pairwise_matches_, cameras_)) {
+        throw std::runtime_error("Could not estimate camera homography");
```

Comment:
You should wrap `stitcher.compute()` from your cmd line app in a try-catch block so errors can be reported correctly.